conanfile.py

```diff
@@ -0,0 +1,184 @@
+from conan.tools.microsoft import is_msvc, msvc_runtime_flag
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools import files
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conans import tools
+import os
+
+required_conan_version = ">=1.55.0"
+
+
+class KnowhereConan(ConanFile):
+    name = "knowhere"
+    description = "Knowhere is written in C++. It is an independent project that act as Milvus's internal core"
+    topics = ("vector", "simd", "ann")
+    url = "https://github.com/milvus-io/knowhere"
+    homepage = "https://github.com/milvus-io/knowhere"
+    license = "Apache-2.0"
+
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cuda": [True, False],
+        "with_asan": [True, False],
+        "with_diskann": [True, False],
+        "with_profiler": [True, False],
+        "with_ut": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": False,
+        "with_cuda": False,
+        "with_asan": False,
+        "with_diskann": False,
+        "with_profiler": False,
+        "with_ut": False,
+    }
+
+    exports_sources = (
+        "src/*",
+        "thirdparty/*",
+        "tests/ut/*",
+        "include/*",
+        "CMakeLists.txt",
+        "*.cmake",
+        "conanfile.py",
+    )
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "gcc": "8",
+            "Visual Studio": "16",
+            "clang": "6",
+            "apple-clang": "10",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("boost/1.81.0")
+        self.requires("glog/0.6.0")
+        self.requires("nlohmann_json/3.11.2")
+        self.requires("llvm-openmp/12.0.1")
+        if self.options.with_ut:
+            self.requires("gtest/1.8.1")
```

Comment:
seems previous we use gtest 1.12, AIK gtest<=1.10 may have some compilation problems for the new compiler.